### PR TITLE
Add zNext hardware exploitation to JIT z/codegen

### PR DESCRIPTION
**Pull request by dchopra001**
This PR adds hardware exploitation to the JIT compiler in the z/codegen for zNext. The commits have been authored by @VermaSh @r30shah and myself.

### CODE REVIEW

**Comment by dchopra001** [comment]
Tagging @joransiu @r30shah @VermaSh for review. If you could please make sure I've brought over all your changes correctly that would be great! Thanks!

**Comment by VermaSh** [comment]
`test/functional/JIT_Test/testng.xml` change [1] is missing from this PR. Do we not need to add the new test suite, `TestJavaIntegerAndLongToString.java`?

[1] https://github.ibm.com/runtimes/openj9/compare/ibm_sdk...zartemis "Internal Diff"

**Comment by dchopra001** [comment]
> test/functional/JIT_Test/testng.xml change [1] is missing from this PR. Do we not need to add the new test suite, TestJavaIntegerAndLongToString.java?

I must have missed this change during a rebase. I've added it back here now. Good catch!

**Review by VermaSh — CHANGES_REQUESTED** [review]

File: runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp [hunk 1]
```diff
@@ -496,6 +498,13 @@
    java_lang_Integer_rotateRight,
    java_lang_Integer_valueOf,
    java_lang_Integer_toUnsignedLong,
+   java_lang_Integer_stringSize,
+   java_lang_Integer_getChars,
+   java_lang_Integer_getChars_charBuffer,
+   java_lang_Integer_toString,
+
```
[the remark this answers is not in the document]
Comment (dchopra001) [reply runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp:505]: Yeah, I would say so. The rest of this file is also formatted in the same way to differentiate between each class' methods that we recognize.

**Comment by VermaSh** [comment]
Looks good, thanks for making the changes @dchopra001!

**Review by r30shah — APPROVED** [review]
LGTM, seems like one of the files needed copyrights updated.

**Review by VermaSh — APPROVED** [review]

**Comment by joransiu** [comment]
For the copyright check, I think we need to update the year for all the files modified with 2022.

**Comment by joransiu** [comment]
jenkins test zlinux jdk8,jdk11,jdk17 depends https://github.com/eclipse/omr/pull/6276

**Comment by joransiu** [comment]
jenkins test sanity zlinux jdk8,jdk11 depends eclipse/omr#6276

**Comment by dchopra001** [comment]
The tests passed and I've now updated the copyright info. Once we have the OMR PR merged, we should be able to merge this as well.

**Review by joransiu — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Dependent upstream PR has merged.  It needs to pass the acceptance build before this PR can merge.

**Comment by joransiu** [comment]
jenkins test sanity zlinux jdk8

**Pull request merged**